File: frontend/src/lib/api/types/notifications.ts
/**
 * GrowFlow — Notification API Types (Phase 8 Batch 3)
 */

export interface NotificationItem {
  id: string;
  user_id: string;
  actor_id?: string | null;
  actor_role?: string | null;
  title: string;
  message: string;
  notification_type: string;
  category: string;
  resource_type?: string | null;
  resource_id?: string | null;
  link: string;
  is_read: boolean;
  read_at?: string | null;
  created_at: string;
}

export interface UnreadCountResponse {
  unread_count: number;
}

export interface MarkAllReadResponse {
  marked_count: number;
}

File: backend/app/api/schemas/notifications.py
"""
GrowFlow — Notification API Request & Response Schemas (Batch 08 / Gate 13).
"""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class NotificationResponseSchema(BaseModel):
    """Response schema for a user notification."""

    id: str
    user_id: str
    actor_id: str | None = None
    actor_role: str | None = None
    title: str
    message: str
    notification_type: str
    category: str
    resource_type: str | None = None
    resource_id: str | None = None
    link: str
    is_read: bool
    read_at: datetime | None = None
    created_at: datetime


class UnreadCountResponseSchema(BaseModel):
    """Response schema for unread count badge."""

    unread_count: int


class MarkAllReadResponseSchema(BaseModel):
    """Response schema after marking all notifications read."""

    marked_count: int
